hermes-consumers/src/test/groovy/pl/allegro/tech/hermes/consumers/consumer/result/undelivered/LogUndeliveredMessageHandlerSpec.groovy

```diff
@@ -0,0 +1,58 @@
+package pl.allegro.tech.hermes.consumers.consumer.result.undelivered
+
+import pl.allegro.tech.hermes.api.Subscription
+import pl.allegro.tech.hermes.common.config.ConfigFactory
+import pl.allegro.tech.hermes.common.config.Configs
+import pl.allegro.tech.hermes.common.exception.InternalProcessingException
+import pl.allegro.tech.hermes.common.kafka.KafkaTopicName
+import pl.allegro.tech.hermes.common.kafka.offset.PartitionOffset
+import pl.allegro.tech.hermes.common.message.undelivered.UndeliveredMessageLog
+import pl.allegro.tech.hermes.consumers.consumer.Message
+import pl.allegro.tech.hermes.consumers.consumer.sender.MessageSendingResult
+import spock.lang.Specification
+
+import java.nio.charset.StandardCharsets
+import java.time.Clock
+
+import static pl.allegro.tech.hermes.api.SentMessageTrace.createUndeliveredMessage
+import static pl.allegro.tech.hermes.consumers.consumer.sender.MessageSendingResult.failedResult
+import static pl.allegro.tech.hermes.consumers.test.MessageBuilder.withTestMessage
+
+
+class LogUndeliveredMessageHandlerSpec extends Specification {
+
+    private static final String MESSAGE_CONTENT = "ala ma kota"
+    private static final long TIME = 732648
+    private static final String CLUSTER = "test_cluster"
+    private static final long OFFSET = 124412L;
+    private static final int PARTITION = 5;
+
+    ConfigFactory configFactory = Stub()
+    Clock clock = Stub()
```

Comment:
clock has static methods to create an instance with given time, no need to stub it